lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -697,6 +697,84 @@ static LogicalResult legalizeHWModule(Block &block,
   // True if these operations are in a procedural region.
   bool isProceduralRegion = block.getParentOp()->hasTrait<ProceduralRegion>();
 
+  // Lower HW wires into SV wires in the appropriate spots. Try to put the
+  // declaration close to the first use in the block, and the assignment right
+  // after the assigned value becomes available.
+  struct WireLowering {
+    hw::WireOp wireOp;
+    Block::iterator declarePoint;
+    Block::iterator assignPoint;
+  };
+  SmallVector<WireLowering> wireLowerings;
+
+  for (auto hwWireOp : block.getOps<hw::WireOp>()) {
+    // Find the earliest point in the block at which the input operand is
+    // available. The assign operation will have to go after that to not create
+    // a use-before-def situation.
+    Block::iterator assignPoint = block.begin();
+    if (auto *defOp = hwWireOp.getInput().getDefiningOp())
+      if (defOp && defOp->getBlock() == &block)
```

Comment:
Is it possible for assignPoint to be null, for example if the wire input is a block argument? Would that be an issue? Maybe it isn't, since I see there is a test for this case.

Reply:
It can't be null. In the worst case when the input is a block argument, the assignPoint is `block.begin()`, which will always be a valid insertion position.